joomla/joomla-cms
# [4.0] password field as custom element

Pull Request for Issue # .

### Summary of Changes
Fields with custom javascript functionality **should** always be custom elements!!!

So, this PR is doing that and also:
- removes the dependency from Bootstrap
- removes the dependency from Font awesome
- Should be accessible
- Should be extremely easy to customise
- Proper implementation of custom elements (eg works asynchronously)


### Testing Instructions
Apply patch and check the
- Login page in the back end
- The login module in the front end
- The edit user page in the backend

### Preview
Backend login: 
![screenshot 2018-12-01 at 15 41 23](https://user-images.githubusercontent.com/3889375/49329471-a7d1db00-f57f-11e8-8bac-c283a0230eaa.png)

Frontend Login:
![screenshot 2018-12-01 at 15 41 43](https://user-images.githubusercontent.com/3889375/49329473-b0c2ac80-f57f-11e8-91e5-62d789b55c08.png)

Edit user page: 
![screenshot 2018-12-01 at 15 41 12](https://user-images.githubusercontent.com/3889375/49329478-c0da8c00-f57f-11e8-8602-89063d5e9acb.png)
 
### Expected result



### Actual result



### Documentation Changes Required


Some Notes here: 

The input prepend/append thingy from the Bootstrap is utterly crap for 2 reasons:
- if used without a proper design language (which of course such thing doesn't exist for Joomla, although I was screaming for it) then the UX is getting inconsistent but more importantly is becoming harder for people to familiarise with all the extra (unneeded) graphics.

- the markup is also crap, eg a span element cannot be focusable (unless if you start doing things like type=button, etc) neither does implements all the built in keyboard functionality. 

All and all: **do not use the input-group** markup/classes just for decoration purposes!!!!


For those really unfamiliar with design language and the necessity for such thing established for Joomla, please check https://design.trello.com/components/form or https://github.com/alexpate/awesome-design-systems
